Framework/Kernel/src/UsageService.cpp

```diff
@@ -27,7 +36,7 @@ Kernel::Logger g_log("UsageServiceImpl");
 //----------------------------------------------------------------------------------------------
 /** FeatureUsage
  */
-FeatureUsage::FeatureUsage(const std::string &type, const std::string &name,
+FeatureUsage::FeatureUsage(const FeatureType &type, const std::string &name,
```

Comment:
This was written pre-C++11 and the advice now for objects that are expensive to copy but cheap to move is that only constructors should take them by value and move them into place:
```c++
 FeatureUsage::FeatureUsage(const FeatureType &type, std::string name,const bool internal) : type(type), name(std::move(name)), internal(internal) {}
```
This covers both cases of having an lvalue and an rvalue reasonably efficiently.